src/UsgsAstroLsSensorModel.cpp

```diff
@@ -108,7 +109,9 @@ void UsgsAstroLsSensorModel::updateState()
    _data.m_HalfSwath = _data.m_Gsd * _data.m_TotalSamples / 2.0;
 
    // Compute half time duration
-   _data.m_HalfTime = _data.m_IntTime * _data.m_TotalLines / 2.0;
+   double fullImageTime = _data.m_IntTimeStartTimes.back() - _data.m_IntTimeStartTimes.front()
+                          + _data.m_IntTimes.back() * (_data.m_TotalLines - _data.m_IntTimeLines.back());
+   _data.m_HalfTime = fullImageTime / 2.0;

```

Comment:
Depending on what the half time actually is, this could be wrong. I assumed half time is the center time in the image because the time at the center pixel is covered by center ephemeris time. For a fixed rate line scan camera, those two values are identical.

Reply:
I also don't know.  Since we have a working HRSC, I assume that this is correct?